given one element size 100% display a circle

kevkevinpal opened the pull request:
Had issue that if we only had one value that was 100% it would still display a divider and now it will just display a circle if we only have one element of 100%

kevkevinpal committed:
given one element size 100% display a circle

Had issue that if we only had one value that was 100% it would still display a divider and now it will just display a circle if we only have one element of 100%

zgordon01 commented:
I'd bump the package version to `1.1.1` in package.json and in the readme

zgordon01 commented on src/Pie.js:
```diff
@@ -41,10 +41,9 @@ const RoundDividers = ({ paintedSections, dividerSize, width, radius, background
-    if (shouldShowDividerForSection) {
+    const { color, startAngle } = section;
+    
+    if (paintedSections.length > 1) {
```
We should still have the dividerSize check here otherwise it will still do this when `dividerSize` is 0 (no dividers). For simplicity see my below comment

milindagrawal replied:
This condition should be checked before looping through `paintedSections`.

zgordon01 commented on src/Pie.js:
```diff
@@ -77,15 +76,15 @@ const RoundDividers = ({ paintedSections, dividerSize, width, radius, background
 
 const getArcAngle = (percentage) => percentage / 100 * 360;
-const shouldShowDivider = (percentage, dividerSize) => percentage <= 100 && !Number.isNaN(dividerSize);
+const shouldShowDivider = (sections) => sections.length > 1;
```
The `!Number.isNaN(dividerSize)` check was originally here for the above comment I made so it might be needed back

milindagrawal replied:
If we are handling it here then we don't need any code changes inside `RoundDivider` function.

zgordon01 commented on src/Pie.js:
```diff
@@ -77,15 +76,15 @@ const RoundDividers = ({ paintedSections, dividerSize, width, radius, background
   let startValue = 0;
   let paintedSections = [];
-
+  const shouldShowDividerForSection = shouldShowDivider(sections);
```
Can rename to `showDividers` or something because this isn't per section anymore

kpallik committed:
Patch 1 (#1)

* moved check for painted sections length above for each so we dont itterate

* changed shouldShowDividerForSection to showDividers

kpallik committed:
Patch 1 (#4)

* moved check for painted sections length above for each so we dont itterate

* changed shouldShowDividerForSection to showDividers

* added new function for rendering our sections and checked to see if we had no values then we just render a grey circle

* bumped package.json to version 1.1.1

* bumped readme to version 1.1.1

kpallik committed:
messed up merge here my second try (#5)

* bumped version to 1.1.1 (#1)

* updated version and fixed 100% issue (#2)

* given one element size 100% display a circle

Had issue that if we only had one value that was 100% it would still display a divider and now it will just display a circle if we only have one element of 100%

* Bumped version to 1.1.1

Co-authored-by: Kevin Pallikunnel <oapallikunnel@gmail.com>

Co-authored-by: Kevin Pallikunnel <oapallikunnel@gmail.com>

kpallik committed:
checking id divider size is NAN and updated readme back to correct versions (#6)

* bumped version to 1.1.1 (#1)

* updated version and fixed 100% issue (#2)

* given one element size 100% display a circle

Had issue that if we only had one value that was 100% it would still display a divider and now it will just display a circle if we only have one element of 100%

* Bumped version to 1.1.1

Co-authored-by: Kevin Pallikunnel <oapallikunnel@gmail.com>

Co-authored-by: Kevin Pallikunnel <oapallikunnel@gmail.com>

kpallik committed:
Updated Readme (#7)

* bumped version to 1.1.1 (#1)

* updated version and fixed 100% issue (#2)

* given one element size 100% display a circle

Had issue that if we only had one value that was 100% it would still display a divider and now it will just display a circle if we only have one element of 100%

* Bumped version to 1.1.1

Co-authored-by: Kevin Pallikunnel <oapallikunnel@gmail.com>

* changed readme version back

Co-authored-by: Kevin Pallikunnel <oapallikunnel@gmail.com>

kpallik committed:
Fix small errors (#8)

* bumped version to 1.1.1 (#1)

* updated version and fixed 100% issue (#2)

* given one element size 100% display a circle

Had issue that if we only had one value that was 100% it would still display a divider and now it will just display a circle if we only have one element of 100%

* Bumped version to 1.1.1

Co-authored-by: Kevin Pallikunnel <oapallikunnel@gmail.com>

* changed readme version back

* fixed small duplicates

Co-authored-by: Kevin Pallikunnel <oapallikunnel@gmail.com>
